src/vivarium/framework/results/observation.py

```diff
@@ -35,6 +37,21 @@ class BaseObservation(ABC):
     results_updater: Callable[[pd.DataFrame, pd.DataFrame], pd.DataFrame]
     results_formatter: Callable[[str, pd.DataFrame], pd.DataFrame]
     stratifications: Optional[Tuple[str]]
+    to_observe: Callable[[Event], bool]
+
+    def gather_results(
```

Comment:
Having methods called `gather_results` and `results_gatherer` is very confusing. We need better names for these methods.

Reply:
Agreed - renamed to this to simply "observe", i.e. 

```
for observation in observations:
    yield observation.observe(event, pop, stratificatons) ...
    ...
```